tests/playwright/utils/SessionManager.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2025 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+import { CsrfManager } from './CsrfManager';
+import { APIRequestContext } from 'playwright/test';
+
+export class SessionManager
+{
+    private request: APIRequestContext;
+    private token: string|null = null;
+
+    public constructor(request: APIRequestContext) {
+        this.request = request;
+    }
+
+    public async changeProfile(profile: string)
+    {
+        const Csrf = new CsrfManager(this.request);
+
+        if (this.token === null) {
+            this.token = await Csrf.getToken();
+        }
+
+        // TODO: add some caching mechanism to avoid loading a profile that is
+        // already being used.
+        const profiles = new Map([
+            ['Self-Service', 1],
+            ['Observer',     2],
+            ['Admin',        3],
+            ['Super-Admin',  4],
+            ['Hotliner',     5],
+            ['Technician',   6],
+            ['Supervisor',   7],
+            ['Read-Only',    8],
+        ]);
+        const profile_id = profiles.get(profile);
+        if (profile_id === undefined) {
+            throw new Error(`Unknown profile: ${profile}`);
+        }
+
+        await this.request.post(
```

Comment:
Should be AJAX rather than a fake form submit?

Reply:
This is an AJAX `POST` request, not sure what mean here.
If by "fake subit" you refer to the CSRF token, it is required since we changed this endpoint to accept only `POST` queries on GLPI's 11.

Reply:
Why is it not just a `$.ajax` call? We already inject the `X-Glpi-Csrf-Token` header with these calls.

If this is how it was done in Cypress (and the taking CSRF tokens from another page), I probably missed the change in a review or I wasn't looking at Cypress changes at the time.

Our CSRF implementation is weird. Idk why there are per-form tokens and per-page reusable token which can all be used for whatever we want on any page.

Reply:
Glpi's `$.ajax` load a CSRF token by extracting a token from the current page (parsing the DOM, same as we are doing here).

It works for GLPI because there is always by definition a page already loaded.

From the context of the tests you may not be on a page yet and thus this process need to be independent.
This is not some GLPI's javascript code attached to a page, this is a dedicated process running on a nodejs script.

Reply:
OK. I understand now. I don't  have any idea to do it different right now, so it seems OK as it is.